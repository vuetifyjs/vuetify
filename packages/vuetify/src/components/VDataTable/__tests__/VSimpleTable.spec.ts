// @ts-nocheck
/* eslint-disable */

// import VSimpleTable from '../VSimpleTable'
import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'

describe.skip('VSimpleTable.ts', () => {
  type Instance = InstanceType<typeof VSimpleTable>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(VSimpleTable, options)
    }
  })

  it('should render', () => {
    const wrapper = mountFunction({
      slots: {
        default: `
          <tr><th>Foo</th><th>Bar</th></tr>
          <tr><td>baz</td><td>qux</td></tr>
        `,
      },
    })

    expect(wrapper.findAll('.v-data-table')).toHaveLength(1)
    expect(wrapper.findAll('.v-data-table .v-data-table__wrapper')).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with custom wrapper', () => {
    const wrapper = mountFunction({
      slots: {
        wrapper: `
          <table>
            <tr><th>Foo</th><th>Bar</th></tr>
            <tr><td>baz</td><td>qux</td></tr>
          </table>
        `,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with top & bottom slots', () => {
    const wrapper = mountFunction({
      slots: {
        top: '<div class="top">Header</div>',
        bottom: '<div class="bottom">Footer</div>',
      },
    })

    expect(wrapper.findAll('.top')).toHaveLength(1)
    expect(wrapper.findAll('.bottom')).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with custom height', () => {
    const wrapper = mountFunction({
      slots: {
        default: `
          <tr><th>Foo</th><th>Bar</th></tr>
          <tr><td>baz</td><td>qux</td></tr>
        `,
      },
      propsData: {
        height: 1000,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should compute classes', () => {
    const wrapper = mountFunction()

    wrapper.setProps({
      dense: true,
    })
    expect(wrapper.vm.classes).toMatchObject({
      'v-data-table--dense': true,
    })
    wrapper.setProps({
      dark: true,
    })
    expect(wrapper.vm.classes).toMatchObject({
      'theme--dark': true,
      'theme--light': false,
    })
    wrapper.setProps({
      fixedHeader: true,
    })
    expect(wrapper.vm.classes).toMatchObject({
      'v-data-table--fixed-header': true,
    })
    wrapper.setProps({
      fixedHeader: false,
      height: 1000,
    })
    expect(wrapper.vm.classes).toMatchObject({
      'v-data-table--fixed-height': true,
    })
  })

  it('should compute classes with top & bottom slots', () => {
    const wrapper = mountFunction({
      slots: {
        top: '<div class="top">Header</div>',
        bottom: '<div class="bottom">Footer</div>',
      },
    })

    expect(wrapper.vm.classes).toMatchObject({
      'v-data-table--has-top': true,
      'v-data-table--has-bottom': true,
    })
  })
})
