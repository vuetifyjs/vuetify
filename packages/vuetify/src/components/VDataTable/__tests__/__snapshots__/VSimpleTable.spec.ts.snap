// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSimpleTable.ts should render 1`] = `
<div class="v-data-table theme--light">
  <div class="v-data-table__wrapper">
    <table>
      <tr>
        <th>
          Foo
        </th>
        <th>
          Bar
        </th>
      </tr>
      <tr>
        <td>
          baz
        </td>
        <td>
          qux
        </td>
      </tr>
    </table>
  </div>
</div>
`;

exports[`VSimpleTable.ts should render with custom height 1`] = `
<div class="v-data-table v-data-table--fixed-height theme--light">
  <div class="v-data-table__wrapper"
       style="height: 1000px;"
  >
    <table>
      <tr>
        <th>
          Foo
        </th>
        <th>
          Bar
        </th>
      </tr>
      <tr>
        <td>
          baz
        </td>
        <td>
          qux
        </td>
      </tr>
    </table>
  </div>
</div>
`;

exports[`VSimpleTable.ts should render with custom wrapper 1`] = `
<div class="v-data-table theme--light">
  <table>
    <tr>
      <th>
        Foo
      </th>
      <th>
        Bar
      </th>
    </tr>
    <tr>
      <td>
        baz
      </td>
      <td>
        qux
      </td>
    </tr>
  </table>
</div>
`;

exports[`VSimpleTable.ts should render with top & bottom slots 1`] = `
<div class="v-data-table v-data-table--has-top v-data-table--has-bottom theme--light">
  <div class="top">
    Header
  </div>
  <div class="v-data-table__wrapper">
    <table>
    </table>
  </div>
  <div class="bottom">
    Footer
  </div>
</div>
`;
